packages/playwright-core/src/server/trace/recorder/snapshotter.ts

```diff
@@ -105,8 +105,13 @@ export class Snapshotter {
   }
 
   async captureSnapshot(page: Page, callId: string, snapshotName: string, element?: ElementHandle): Promise<void> {
+
+    const noscript_remove = this._context._options.javaScriptEnabled
```

Comment:
We can add a second parameter to [`frameSnapshotStreamer` here](https://github.com/microsoft/playwright/blob/main/packages/playwright-core/src/server/trace/recorder/snapshotter.ts#L89) to make it more straightforward.

```ts
    const removeNoScript = !!this._context._options.javaScriptEnabled;
    const initScript = `(${frameSnapshotStreamer})("${this._snapshotStreamer}", ${removeNoScript})`;
```

Reply:
Yes -- such a modification would be very useful!

But this change should be better written by someone much more familiar with this code base.
I really couldn't grasp all lifecycle concerns and context isolation efforts hidden behind some strange looking passages of the sources.